modules/boltlib/lib/puppet/functions/run_script.rb

```diff
@@ -10,26 +10,16 @@
     type 'TargetOrTargets = Variant[String[1], Target, Array[TargetOrTargets]]'
   end
 
-  dispatch :run_script_with_args do
-    scope_param
-    param 'String[1]', :script
-    param 'TargetOrTargets', :targets
-    param 'Struct[arguments => Array[String]]', :arguments
-    return_type 'ExecutionResult'
-  end
-
   dispatch :run_script do
     scope_param
     param 'String[1]', :script
-    repeated_param 'TargetOrTargets', :targets
+    param 'TargetOrTargets', :targets
+    optional_param 'Hash[String[1], Any]', :options
```

Comment:
It's not clear to me if this changes behavior. Before you needed to do
`run_script('script', $targets, arguments => ['-a', '-b'])`

The same would still work then, because you look for `arguments` in the options hash.